pkg/util/nichelper/helper.go

```diff
@@ -67,3 +80,45 @@ func IdentifyManagementNIC() ([]string, error) {
 	logrus.Debugf("skipping interfaces with pciAddresses: %v", pciAddresses)
 	return pciAddresses, nil
 }
+
+// identifyClusterNetworks will identify vlanConfigs covering the current node and identify NICs in use for
+// vlanconfigs
+func identifyClusterNetworks(nodeName string, nodeCache ctlcorev1.NodeCache, vlanConfigCache ctlnetworkv1beta1.VlanConfigCache) ([]string, error) {
+	var nicsList []string
+	vlanConfigList, err := vlanConfigCache.List(labels.NewSelector())
+	if err != nil {
+		return nil, fmt.Errorf("error fetching vlanconfigs: %v", err)
+	}
+	for _, v := range vlanConfigList {
+		ok, err := currentNodeMatchesSelector(nodeName, nodeCache, v.Spec.NodeSelector)
+		if err != nil {
+			return nil, fmt.Errorf("error evaulating nodes from selector: %v", err)
+		}
+		if ok {
+			nicsList = append(nicsList, v.Spec.Uplink.NICs...)
+		}
+	}
+	return nicsList, nil
+}
+
+// currentNodeMatchesSelector will use the label selectors from VlanConfig to identify if node is
+// in the matching the VlanConfig
+func currentNodeMatchesSelector(nodeName string, nodeCache ctlcorev1.NodeCache, vlanConfigLabels map[string]string) (bool, error) {
```

Comment:
The vlanconfig has an annotation network.harvesterhci.io/matched-nodes which records the matching nodes. The harvester network webhook adds the annotation automatically. It would be more convenient to use the annotation.

Reply:
does the webhook update the annotation when nodes matching label selector are added to the cluster?

Reply:
Yes.